### PR TITLE
map and expose miscellaneous fees to CCR bills

### PR DESCRIPTION
**Pull request by jsugarman**
**What**
Add CCR miscellaneous fee equivalents to the JSON api output 

**Why**
For injection of equivalent CCCD fees into CCR.

**NOTE**
CCR miscellaneous fees are not synonymous with CCCD miscellaneous fees. Some
Basic Fees and Fixed fees in CCCD are handled as miscellaneous fees in CCR.

### CODE REVIEW

File: app/interfaces/api/entities/ccr/adapted_advocate_fee.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        expose :case_numbers
+
+        # NOTE: for possible use in comparisons between CCCD and CCR calculation comparision
+        # expose :calculated_fee, as: :calculatedFee
+
+        private
+
+        def quantity
+          1.0
```
Comment (colinbruce) [inline app/interfaces/api/entities/ccr/adapted_advocate_fee.rb:27]: Why deliberately set this to 1?  Wouldn't the app/api force a value?

Or, does the fact that the fee exist often mean the quantity is one?

I'll keep reading... :)

File: config/schemas/ccr_claim_schema.json [hunk 2]
```diff
@@ -95,40 +95,56 @@
           },
           "number_of_cases": {
             "id": "/properties/bills/items/properties/number_of_cases",
-            "type": "integer"
+            "type": "string"
```
Comment (colinbruce) [inline config/schemas/ccr_claim_schema.json:98]: Is this because CCR needs strings?

**Review by colinbruce — APPROVED** [review]
A couple of small questions, but if the answers are obvious, feel free to merge.

File: app/interfaces/api/entities/ccr/adapted_advocate_fee.rb (hunk 1 above)
Comment (jsugarman) [reply app/interfaces/api/entities/ccr/adapted_advocate_fee.rb:27]: In CCR there is only ever one advocate fee. Ideally CCR should just assume this and the other hardcoded values and we can then do away with exposing quantity, rate and amount entirely. 

will speak to @cibuanokpe about this, but this has been going exposed for a while now and am just exposing the same thing in a different way here.

File: config/schemas/ccr_claim_schema.json (hunk 2 above)
Comment (jsugarman) [reply config/schemas/ccr_claim_schema.json:98]: Had a long slack chat with clem about this and we have settled on using strings for any number. BigDecimals, in particular, default to strings via the BigDecimal.to_json method. There is quite alot of chatter about this stuff e.g. https://github.com/rails/rails/issues/25017 . rails used to have an override but it was removed for sensible reasons. I made integers strings too firstly for consistency (CCR can just assume strings and handled appropriately) but also was reading that the JSON datatype integer is not parsed consistently by some libraries in any event - meaning you could put a float/decimal in and might get away with it.

Comment (colinbruce) [reply config/schemas/ccr_claim_schema.json:98]: 👍

File: app/interfaces/api/entities/ccr/adapted_advocate_fee.rb (hunk 1 above)
Comment (colinbruce) [reply app/interfaces/api/entities/ccr/adapted_advocate_fee.rb:27]: :+1:

**Comment by colinbruce** [comment]
I restarted that build after a flickering test... Other than than that good to merge 👍

**Pull request merged**